packages/devtools_app/lib/src/screens/network/network_controller.dart

```diff
@@ -56,6 +59,34 @@ class NetworkController extends DisposableController
     );
     subscribeToFilterChanges();
   }
+  List<DartIOHttpRequestData>? _httpRequests;
+
+  String? exportAsHarFile() {
+    final exportController = ExportController();
+
+    _httpRequests =
+        filteredData.value.whereType<DartIOHttpRequestData>().toList();
+
+    if (_httpRequests!.isEmpty) {
+      debugPrint('No valid request data to export');
+      return '';
+    }
+
+    try {
+      if (_httpRequests != null && _httpRequests!.isNotEmpty) {
+        // Build the HAR object
+        final har = buildHar(_httpRequests!);
```

Comment:
Now that we have the HarNetworkData class, this can be `final har = HarNetworkData(_httpRequests);` and then below `json.encode(har.toJson()`.

Reply:
done